### ISSUE TITLE
Add support for DataScroller in mobile render kit

### ISSUE THREAD

**Issue by JohnSchneider** [primefaces/primefaces#140]
The new pagination support for mobile DataGrid and DataList is nice, but sometimes DataScroller is a better fit in a mobile UI.  Please add support for DataScroller in mobile render kit.

**Comment by cagataycivici** [comment]
Sounds good to me, if we can find a way to hook-in to scrolling via touch, we'll be able to do it. Need to do some research. Any tips are appreciated.

**Comment by JohnSchneider** [comment]
I've used JQuery plugin TouchCarousel for really nice horizontal swipe functionality.  However, I don't think it supports vertical scrolling like we have with DataScroller.  Keeping with JQuery, I think dragendJS might be the best option for vertical scolling. http://stereobit.github.io/dragend/

There is also SlideToucher, but I'm not sure how mature it is at this point. The demo wasn't working for me at all on a desktop. https://github.com/Yuripetusko/slideToucher

**Comment by cagataycivici** [comment]
Thank you for the feedback, I'll review these. First choice is creating our own implementation but if it is complex then we'll probably use a 3rd party.

**Comment by JohnSchneider** [comment]
Pending an official feature addition, I've implemented my own PF mobile data scrolling, based on the JQuery Mobile "scrollstop" event, with no other dependencies.  Maybe this could serve as an idea for actual PF implementation.

JavaScript:

```
$(document).ready(function() {
    // Handle lazy loading products list on scroll to bottom
    if ("#products-list".length) {
        // using document object, but could use a scrollable component instead
        $(document).on("scrollstop", function () {
            // param: 300 px from bottom buffer; should be a PF component attribute
            if (isAtBottom(300)) { 
                loadProducts(); // the name of the PF remoteCommand
            }
        });
    }
});
// taken thankfully from http://jsfiddle.net/shanabus/LJTJt/1/
function isAtBottom(buffer){
    var totalHeight, currentScroll, visibleHeight;
    if (document.documentElement.scrollTop)
        currentScroll = document.documentElement.scrollTop;
    else
        currentScroll = document.body.scrollTop;

    //totalHeight = document.body.offsetHeight;
    totalHeight = $(document).height();
    visibleHeight = document.documentElement.clientHeight;

    return (totalHeight <= currentScroll + visibleHeight + bufffer)
}
```

JSF/PF XHTML:

```
<h:form id="load-products-form" prependId="false">
  <p:remoteCommand id="load-products-command" name="loadProducts"
    action="#{browsingController.loadProducts}" update=":products-panel"
    async="false" />
</h:form>
<p:outputPanel id="products-panel">
  <comp:products id="products"
    value="#{browsingController.displayedProducts}"
    showFilter="#{browsingController.inactiveProductsFound}" />
</p:outputPanel>
```

Superclass of CDI Bean:

```
public abstract class ProductListController implements Serializable {

    private static final long serialVersionUID = 1L;
    private static final int pageSize = 18;

    private List<Product> products;
    private List<Product> displayedProducts;
    private int pages;
    private int pagesLoaded;

    public void loadProducts() {
        RequestContext context = RequestContext.getCurrentInstance();
        if (pagesLoaded != pages) {
            int toIndex = (pagesLoaded + 1) * pageSize < products.size()
                    ? (pagesLoaded + 1) * pageSize
                    : products.size();
            List<Product> products = this.products.subList(pagesLoaded
                    * pageSize, toIndex);
            displayedProducts.addAll(products);
            pagesLoaded++;
            if (pagesLoaded == pages)
                context.execute("$(document).off('scrollstop')");
        }
        context.execute("$('#products-panel').trigger('create')");
    }

    public List<Product> getDisplayedProducts() {
        return displayedProducts;
    }

    public int getPages() {
        return pages;
    }

    protected List<Product> getProducts() {
        return products;
    }

    protected void setProducts(List<Product> products) {
        this.products = products;
        pages = products.size() % pageSize == 0
                ? products.size() / pageSize 
                : (products.size() / pageSize) + 1;
                displayedProducts = new ArrayList<>();
        loadProducts();
    }

}
```

**Comment by tandraschko** [comment]
PrimeFaces Mobile will be deprecated  in 6.2 and removed in 6.3 ( see https://www.primefaces.org/primefaces-6-2-roadmap/)
Therefor i will close this issue as won't fix. If you will provide a PR to fix this issue until 6.2 RC1, i can merge it.
